src/common/utils/gpo.h

```diff
@@ -61,6 +62,37 @@ namespace powertoys_gpo {
 
     // The registry value names for other PowerToys policies.
     const std::wstring POLICY_ALLOW_EXPERIMENTATION = L"AllowExperimentation";
+    const std::wstring POLICY_CONFIGURE_ENABLED_POWER_LAUNCHER_ALL_PLUGINS = L"PowerLauncherAllPluginsEnabledState";
+
+
+    inline std::optional<std::wstring> ReadRegistryStringValue(HKEY hKey, const std::wstring& registry_value_name)
+    {
+        DWORD reg_value_type = REG_SZ;
+
+        DWORD string_buffer_capacity;
+        // Request required buffer capacity / string length
+        if (RegQueryValueExW(hKey, registry_value_name.c_str(), 0, nullptr, nullptr, &string_buffer_capacity) != ERROR_SUCCESS)
+        {
+            return std::nullopt;
+        }
+        else if (string_buffer_capacity == 0)
+        {
+            return std::nullopt;
+        }
+
+        // int b_size = string_buffer_capacity / sizeof(WCHAR);
+        string_buffer_capacity = 1024;
+        wchar_t string_value[1024]{};
```

Comment:
@stefansjfw 
Is it possible to create the array at runtime with the size from the variable `b_size`?

Reply:
@jaimecbernardo
Can you help here? Currently it is not working without a fixed array definition.

Reply:
Note to me: Copy code solution from BugReportTool: https://github.com/microsoft/PowerToys/blob/main/tools/BugReportTool/BugReportTool/RegistryUtils.cpp#L269-L271

Reply:
Did you end up figuring it out @htcfreek ? Getting what size you need first would be the way to go indeed.

Reply:
The problem is not to get the required size first. But I need to create the array of type wchar*. And if I use this typ the array is broken after reading the value. But I will try to go with the solution from bugreport tool which seems to work there perfectly.